packages/SwingSet/src/kernel/kernel.js

```diff
@@ -557,13 +560,13 @@ export default function buildKernel(kernelEndowments) {
    * might tell the manager to replay the transcript later, if it notices
    * we're reloading a saved state vector.
    */
-  function addVatManager(vatID, manager, options) {
+  function addVatManager(vatID, manager, vatOptions, creationOptions) {
```

Comment:
I don't think this should take `vatOptions`: if those are the parameters that will get passed into `buildRootObject`'s second argument, then they have to be supplied to the function that creates the manager originally. `addVatManager` takes a manager that's already been built and wires it into the delivery/syscall pathways, but by that point it's too late to provide anything to `buildRootObject`.

(also, it doesn't look like `vatOptions` are used within this function, which should have triggered a lint error)

`creationOptions`, yes.

Reply:
I think you are right.  This may have slipped in during debugging.  There were some earlier intermediate states where lint was complaining, so I'm not sure what's going on with that here.  But I'll take it out.

This points out a coding issue worthy of a longer discussion elsewhere: I think the options bag pattern does not compose well.